_stbt/core.py

```diff
@@ -1191,6 +1191,52 @@ def argparser():
     return parser
 
 
+class FrameObject(object):
+    def __init__(self, frame):
+        property_names = sorted([
+            p for p in dir(self.__class__)
+            if isinstance(getattr(self.__class__, p), property)])
+        if 'is_visible' not in property_names:
+            raise TypeError(
+                "Classes deriving from FrameObject must define a property "
+                "named 'is_visible'")
+        self.__attrs = ["is_visible"] + sorted(
+            x for x in property_names
+            if x != "is_visible" and not x.startswith('_'))
+
+        if frame is None:
+            raise ValueError("FrameObject: frame must not be None")
+        self._frame = frame
+
+    def __repr__(self):
+        if self:
+            args = ", ".join(
+                "%s=%r" % (x, getattr(self, x)) for x in self.__attrs)
+        else:
+            args = "%s=False" % self.__attrs[0]
+
+        return "%s(%s)" % (self.__class__.__name__, args)
+
+    def __nonzero__(self):
+        return bool(getattr(self, self.__attrs[0]))
+
+    def _astuple(self):
+        if self:
+            return tuple(getattr(self, x) for x in self.__attrs)
+        else:
+            return (False,)
+
+    def __cmp__(self, other):
```

Comment:
I'm thinking of these like a lazily evaluated `namedtuple`, so defining `__cmp__` seems natural to me.  Although I think this `__cmp__` could be improved so as to avoid querying all the properties at once invalidating lazyness.